lib/memoryprofiler/derive/tests/basic.rs

```diff
@@ -0,0 +1,23 @@
+use wasmer_memoryprofiler::r#trait::MemoryUsage;
+use wasmer_memoryprofiler_derive::MemoryUsage;
```

Comment:
Mentioned it on slack but I'll say it again here: Rust doesn't care about name collisions with derive macros and traits. In fact it's actually a big feature if they're the same because then you can just `use` one thing and you get both in scope!

Reply:
(Not strictly relevant for this file, but generally it's useful; i.e. the trait shouldn't be publicly exported under `r#trait`)

Reply:
I'm not sure what you're suggesting. Ideally I would like the trait to be `MemoryUsage` so that you can `impl MemoryUsage for Whatever` and also for the proc_macro to be used as `#[derive(MemoryUsage)]`. The trait needs to be public so that people can impl it for their own types. I'm not sure how to arrange it so that you can use one thing and get both types when they have to be in separate crates because proc macro crates much be separate from everything else.